controllers/addon_controller.go

```diff
@@ -123,19 +123,27 @@ func (r *AddonReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 		return reconcile.Result{}, ignoreNotFound(err)
 	}
+	
+	return r.execAddon(ctx, req, log, instance)
+}
 
+func (r *AddonReconciler) execAddon(ctx context.Context, req reconcile.Request, log logr.Logger, instance *addonmgrv1alpha1.Addon) (reconcile.Result, error) {
+	defer func() {
+		if err := recover(); err != nil {
+			log.Info("Error: Panic occurred during execAdd %s/%s due to %s", instance.Namespace, instance.Name, err)
+		}
+	}()
 	// Process addon instance
 	ret, err := r.processAddon(ctx, req, log, instance)
```

Comment:
```suggestion
	ret, procErr := r.processAddon(ctx, req, log, instance)
```

Reply:
this was from the original code but ok I will make change